HW04_code.Rmd

```diff
@@ -0,0 +1,81 @@
+---
+title: "HW04_code"
+author: "Laura Hunter"
+date: "7/27/2020"
+output: git_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Cleaning the Data
+
+My hope with this assignment is to take the "2lot_results.csv" file that is in this directory and change the data in it so that it looks like the "2lot_results_clean.csv" file. Once I've done that, I'll be able to use the code I made from last week to plot Subject 8's data.
+
+```{r data cleaning}
+library(tidyverse) #load tidyverse packages
+
+unclean_2lot <- read.csv("2lot_results.csv", stringsAsFactors = FALSE) #Load raw 2lot data
+
+unclean_2lot <- unclean_2lot %>%
+  filter(timing =="ok") %>% #remove all trials with timing errors
+  select(block_name, left_line1:right_line2, choice) %>% #only cols necessary for this analysis
+  mutate(left_EV = (left_line1 + left_line2)/2) %>% #expected value of left side
+  mutate(left_sigma = left_line2 - left_line1) %>% #sigma of left side
+  mutate(right_EV = (right_line1 + right_line2)/2) %>% #expected value of right side
+  mutate(right_sigma = right_line2 - right_line1) %>% #sigma of right side
+  select(block_name, choice:right_sigma) %>% #no longer need line data
+  mutate(deltaEV = if_else(left_sigma > right_sigma, true = left_EV - right_EV, false = right_EV - left_EV)) %>% #deltaEV is hi sig - low sig
+  mutate(chose_left_hi = if_else((left_sigma > right_sigma) & (choice == "left"), true = T, false = F)) %>% #chose left when left is hi sig
+  mutate(chose_right_hi = if_else((left_sigma < right_sigma) & (choice == "right"), true = T, false = F)) %>% #chose right when right is hi sig
+  mutate(good_choice = chose_left_hi + chose_right_hi) %>% #a good choice is a high choice
+  select(block_name, deltaEV, good_choice) %>% #we don't need the above, transitional columns
+  arrange(block_name, deltaEV)#let's arrange by block and deltaEV
+
+unclean_2lot$good_choice <- factor(unclean_2lot$good_choice, levels = c(0,1)) #factor the good_choice to ensure no data drops
+unclean_2lot$block_name <- factor(unclean_2lot$block_name, levels = c("observe", "guess", "intervene", "observe-repeat")) #level to match clean
+
```

Comment:
It is not very clear for me here what was the purpose of creating the "cleaned_deltaEV" object. This is clarified later, but perhaps a comment here would have been useful to be able to follow the code easier.